### ISSUE TITLE
Analyzers stop working when I set AnalysisLevel to preview

### ISSUE THREAD

**Issue by Neme12** [dotnet/roslyn-analyzers#6909]
### Analyzer source

**SDK**: [Built-in CA analyzers in .NET 5 SDK or later](https://learn.microsoft.com/dotnet/fundamentals/productivity/code-analysis)

**Version**: SDK 8.0.100-preview.7.23376.3

### Describe the bug
With the following .csproj and source file:
```xml
<Project Sdk="Microsoft.NET.Sdk">

  <PropertyGroup>
    <OutputType>Exe</OutputType>
    <TargetFramework>net8.0</TargetFramework>
    <LangVersion>preview</LangVersion>
    <Nullable>enable</Nullable>
    <ImplicitUsings>enable</ImplicitUsings>
    <AnalysisLevel>latest</AnalysisLevel>
    <AnalysisMode>all</AnalysisMode>
  </PropertyGroup>

</Project>
```
```c#
public sealed class CollectionsMarshalPolyfillTest
{
    public void TestAsSpan_Null()
    {
    }
}
```
I get analyzer warnings as expected (in this case, [CA1050](https://learn.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca1050) and [CA1707](https://learn.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca1707).
![image](https://github.com/dotnet/roslyn-analyzers/assets/11444821/608cee39-2ab5-49c1-af98-f9f5c92aaa4a)

However, if I change AnalysisLevel from latest to preview, the warnings disappear, as if AnalysisMode wasn't set to all, even though it is:
![image](https://github.com/dotnet/roslyn-analyzers/assets/11444821/ec3a37dc-8f6f-4ed2-a569-ed03e9cf2103)

**Comment by mavasani** [comment]
Verified that this was fixed with #6944, hence should be fixed in .NET 8 RC2 SDK.

Basically, this repros with https://dev.azure.com/dnceng/public/_artifacts/feed/dotnet8/NuGet/Microsoft.CodeAnalysis.NetAnalyzers/overview/8.0.0-preview.23466.2, but not with https://dev.azure.com/dnceng/public/_artifacts/feed/dotnet8/NuGet/Microsoft.CodeAnalysis.NetAnalyzers/overview/8.0.0-preview.23468.1.  #6944 ensures that we generate the required editorconfig files with new .NET releases.

@carlossanlop @buyaa-n @jeffhandley - we need to ensure that we have process/documentation in place to ensure the following happen:
- For every new release branch that is forked on the roslyn-analyzers repo, we should make a change similar to #6944 to create a new Release section in the AnalyzerReleases.Shipped.md file
- For every new public preview/RC/GA/Patch release of .NET SDK, we should publish the corresponding Microsoft.CodeAnalysis.NetAnalyzers.nupkg on NuGet.org
I'll file a separate tracking issue for this.

**Comment by mavasani** [comment]
@Neme12 Thank you for reporting the issue, much appreciated!
